targets/TARGET_STM/stm_spi_api.c

```diff
@@ -349,22 +350,54 @@ static inline int ssp_busy(spi_t *obj)
 
 int spi_master_write(spi_t *obj, int value)
 {
-    uint16_t size, ret;
-    int Rx = 0;
     struct spi_s *spiobj = SPI_S(obj);
     SPI_HandleTypeDef *handle = &(spiobj->handle);
 
-    size = (handle->Init.DataSize == SPI_DATASIZE_16BIT) ? 2 : 1;
+#ifdef STM32_SPI_KEEP_HAL_USAGE
```

Comment:
Where is this macro coming from ? IS it documented anywhere? This changes the implementation .